src/OctoshiftCLI.Tests/gei/Commands/MigrateSecretScanningAlertsCommandTests.cs

```diff
@@ -0,0 +1,59 @@
+using FluentAssertions;
+using Moq;
+using OctoshiftCLI.Contracts;
+using OctoshiftCLI.GithubEnterpriseImporter;
+using OctoshiftCLI.GithubEnterpriseImporter.Commands;
+using Xunit;
+
+namespace OctoshiftCLI.Tests.GithubEnterpriseImporter.Commands
+{
+    public class MigrateSecretScanningAlertsCommandTests
+    {
+        private readonly Mock<ISourceGithubApiFactory>
+            _mockSourceGithubApiFactory = new Mock<ISourceGithubApiFactory>();
+
+        private readonly Mock<ITargetGithubApiFactory>
+            _mockTargetGithubApiFactory = new Mock<ITargetGithubApiFactory>();
+
+        private readonly Mock<OctoLogger> _mockOctoLogger = TestHelpers.CreateMock<OctoLogger>();
+
+        private readonly Mock<EnvironmentVariableProvider> _mockEnvironmentVariableProvider =
+            TestHelpers.CreateMock<EnvironmentVariableProvider>();
+
+        private readonly Mock<ISecretScanningAlertServiceFactory> _mockSecretScanningAlertServiceFactory =
+            new Mock<ISecretScanningAlertServiceFactory>();
+
+        private readonly MigrateSecretScanningAlertsCommand _command;
+
+        public MigrateSecretScanningAlertsCommandTests()
+        {
+            _command = new MigrateSecretScanningAlertsCommand(
+                _mockOctoLogger.Object,
+                _mockSecretScanningAlertServiceFactory.Object,
+                _mockSourceGithubApiFactory.Object,
+                _mockTargetGithubApiFactory.Object,
+                _mockEnvironmentVariableProvider.Object
+            );
+        }
+
+        [Fact]
+        public void Should_Have_Options()
+        {
+            _command.Should().NotBeNull();
+            _command.Name.Should().Be("migrate-secret-alerts");
+            _command.Options.Count.Should().Be(11);
+
+            TestHelpers.VerifyCommandOption(_command.Options, "github-source-org", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "source-repo", true);
+            TestHelpers.VerifyCommandOption(_command.Options, "github-target-org", true);
+            TestHelpers.VerifyCommandOption(_command.Options, "target-repo", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "target-api-url", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "ghes-api-url", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "no-ssl-verify", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "github-source-pat", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "github-target-pat", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "verbose", false);
+            TestHelpers.VerifyCommandOption(_command.Options, "dry-run", false);
+        }
+    }
```

Comment:
I'd also like to see a test covering a happy path (or a sad path if any) scenario.

Reply:
Ouch, the amount of data required to mock that via the GitHub APIs is pretty heavy to get it all to link up... I will do so but maintaining that test is bound to be pretty horrible if it ever needs changing.

Reply:
Usually it's a code smell whenever we need to mock too much. Let's start with having a test and later on we can refactor in a separate PR if need be.

Reply:
Understood, but its the underlying data from the source and target here needs to be carefully mocked to cover the matching between the secret alerts and the various outcomes that can arise. I can do a trivial one, but might as well do something meaningful if going down this path.

Reply:
> but its the underlying data from the source and target here needs to be carefully mocked to cover the matching between the secret alerts and the various outcomes that can arise

Yes. This is exactly the reason why we add different unit tests and spend some time trying to cover edge cases, etc., which will save us a lot of time in the future in case of any bugs, etc. So I would spend a bit more time writing good tests to test all scenarios to prevent future bugs that could have been caught by tests.